asdf/extension/_legacy.py

```diff
@@ -179,6 +179,16 @@ def get_cached_asdf_extension_list(extensions):
     -------
     asdf.extension.AsdfExtensionList
     """
+    warnings.warn(
+        "get_cached_asdf_extension_list is deprecated. "
+        "Please see the new extension API "
+        "https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html",
+        AsdfDeprecationWarning,
+    )
+    return _get_cached_asdf_extension_list(extensions)
```

Comment:
`_legacy` is already a private module. This function only becomes public when it is added to the interface in `asdf.extension.__init__` why not simply define the deprecated function there and leave `_legacy` alone.

That is add
```python
def get_cached_asdf_extension_list(extensions):
    """
    Get a previously created AsdfExtensionList for the specified
    extensions, or create and cache one if necessary.  Building
    the type index is expensive, so it helps performance to reuse
    the index when possible.
    Parameters
    ----------
    extensions : list of asdf.extension.AsdfExtension
    Returns
    -------
    asdf.extension.AsdfExtensionList
    """
    from ._legacy import get_cached_asdf_extension_list(extensions)
    warnings.warn(
        "get_cached_asdf_extension_list is deprecated. "
        "Please see the new extension API "
        "https://asdf.readthedocs.io/en/stable/asdf/extending/converters.html",
        AsdfDeprecationWarning,
    )
    return get_cached_asdf_extension_list(extensions)
```
to `asdf.extension.__init__` (removing the `get_cached_asdf_extension_list` from the import at the top of `asdf.extension.__init__`.